packages/ui/src/util/bridgeUtils.ts

```diff
@@ -68,6 +72,25 @@ export const populateBridgeTransaction = (
     }
 }
 
+export const hasEnoughFundsToPayTheGasInSendTransaction = (
+    isEIP1559Compatible: boolean,
+    balance: BigNumber,
+    gasPricesData: GasPriceData
+): boolean | undefined => {
+    const gasPrices = getDisplayGasPrices(
```

Comment:
I wouldn't call `getDisplayGasPrices` here, since this function was mostly thought for displaying info in the UI. Also, you only need the `gasPrices.average` calculation.

I would extract the code that checks whether it is `EIP1559Compatible` and invokes `calculateTransactionGas` from `getDisplayGasPrices`.

Imagine you create a function that is named: 
```typescript
function getTransactionFees(isEIP1559Compatible:boolean, feeData:FeeData, estimatedBaseFee:BigNumber, gasLimit: BigNumber){
          let gasPrice : BigNumber | undefined = gasPrice?.gasPrice;
           if (isEIP1559Compatible && estimatedBaseFee) {
                    const baseFee = BigNumber.from(estimatedBaseFee)
                    const priority = BigNumber.from(
                        gasPrice?.maxPriorityFeePerGas ?? 0
                    )
                    gasPrice = baseFee.add(priority)
                }
                calculateTransactionGas(gasLimit,  BigNumber.from(gasPrice ?? 1)),
}
```

Then, you can use the function above in the `getDisplayGasPrices` too. 


And in your code:

```typescript
 const averageTransactionFee = getTransactionFees(isEIP1559Compatible, gasPricesData.gasPricesLevels.average, gasPricesData.estimatedBaseFee!, SEND_GAS_COST);
  if (averageTransactionFee === undefined) {
        return undefined
    }
    return BigNumber.from(balance).gte(
        BigNumber.from(averageTransactionFee)
    )
```